compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -390,7 +390,18 @@ parseProjectConfig project = do
     dependencies <-
         queryProjectConfigRequired ["dependencies"] project
     sdkVersion <- queryProjectConfigRequired ["sdk-version"] project
+    consistencyCheck name version dependencies
     Right $ PackageConfigFields name main exposedModules version dependencies sdkVersion
+  where
+    consistencyCheck n ver deps
+        | n <> "-" <> ver `elem` map takeBaseName deps =
```

Comment:
I don’t think this check is sufficient. You are relying on people using the default DAR name which isn’t a valid assumption (e.g. that wouldn’t have caught my error). I think we should ask GHC for the name of the packages in the package db.

Reply:
I'll close this PR and solve it with a lookup in the package db.

Reply:
Thanks!